compat/proc.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 1996-2022 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_COMPAT_PROC_H
+#define SQUID_COMPAT_PROC_H
+
+#if HAVE_SYS_PRCTL_H
+#include <sys/prctl.h>
+#endif
+#if HAVE_SYS_PROCCTL_H
+#include <sys/procctl.h>
+#endif
+#if HAVE_PRIV_H
+#include <priv.h>
+#endif
+
+inline int
+ProcSetTraceable() {
```

Comment:
This function does not match up with the intended use of libcompat as a substitute source of missing API symbols. eg PR_SET_DUMPABLE and prctl() themselves in this case.

I like your implementation better than what would be needed to fit here in libcompat. So my request is that you move this to a static function in src/tools.cc instead of extending libcompat.

Reply:
ok.